sonic-slave-buster/Dockerfile.j2

```diff
@@ -1,5 +1,7 @@
 {%- if CONFIGURED_ARCH == "armhf" %}
+FROM multiarch/qemu-user-static:x86_64-arm-5.0.0-2 as qemu
 FROM multiarch/debian-debootstrap:armhf-buster
+COPY --from=qemu /usr/bin/qemu-arm-static /usr/bin
```

Comment:
>qemu /usr/bin/qemu-arm-static [](start = 12, length = 29)

Why copy a binary from another image? Seems a hack.

Reply:
Not a hack, but to change the qemu version from 4 to 5, and qemu 4 has severe segfaults with golang

Reply:
Could you achieve this goal by installing qemu-user-static package? I see version 5.0 [https://packages.debian.org/buster-backports/qemu-user-static](https://packages.debian.org/buster-backports/qemu-user-static)

---
In reply to: [526621660](https://github.com/Azure/sonic-buildimage/pull/5749#discussion_r526621660) [](ancestors = 526621660)

Reply:
Couldn't able to install qemu from debian buster. 
Error Logs:
Unpacking qemu-user-static (1:5.0-14~bpo10+1) ...
dpkg (subprocess): unable to execute rm command for cleanup (rm): Too many levels of symbolic links
dpkg: error while cleaning up:
rm command for cleanup subprocess returned error exit status 2
Could not exec dpkg!
E: Sub-process /usr/bin/dpkg returned an error code (100)
E: Problem executing scripts DPkg::Post-Invoke 'rm -f /var/cache/apt/archives/*.deb /var/cache/apt/archives/partial/*.deb /var/cache/apt/*.bin || true'
E: Sub-process returned an error code
The command '/bin/sh -c apt-get install -y qemu-user-static=1:5.0-14~bpo10+1' returned a non-zero code: 100

Reply:
Couldnt able to install qemu from debian buster.
Error Logs:
Unpacking qemu-user-static (1:5.0-14~bpo10+1) ...
dpkg (subprocess): unable to execute rm command for cleanup (rm): Too many levels of symbolic links
dpkg: error while cleaning up:
rm command for cleanup subprocess returned error exit status 2
Could not exec dpkg!
E: Sub-process /usr/bin/dpkg returned an error code (100)
E: Problem executing scripts DPkg::Post-Invoke 'rm -f /var/cache/apt/archives/*.deb /var/cache/apt/archives/partial/*.deb /var/cache/apt/*.bin || true'
E: Sub-process returned an error code
The command '/bin/sh -c apt-get install -y qemu-user-static=1:5.0-14~bpo10+1' returned a non-zero code: 100

Reply:
Not familiar with this error message. I notice you are using `dpkg` directly. Did you try use `apt-get`. Please add something like below into `/etc/apt/sources.list`
```
deb [arch=amdhf] [http://debian-archive.trafficmanager.net/debian](http://debian-archive.trafficmanager.net/debian) buster-backports main
```
I don't have an armhf dev environment, so please test it first.

---
In reply to: [530707840](https://github.com/Azure/sonic-buildimage/pull/5749#discussion_r530707840) [](ancestors = 530707840,526621660)

Reply:
yes, I have already done that.
_echo 'deb [arch=armhf] http://ftp.debian.org/debian buster-backports main' >> /etc/apt/sources.list
apt-get install -y qemu-user-static=5.0-14~ bpo10+1_

Reply:
Sorry, I see `deb ...` line, but I cannot find the second part.
```
apt-get install -y qemu-user-static=5.0-14~ bpo10+1
```

---
In reply to: [532104630](https://github.com/Azure/sonic-buildimage/pull/5749#discussion_r532104630) [](ancestors = 532104630,530707840,526621660)

Reply:
I tried below lines and it was not working. got same errors as above. So, I didnt add this to the PR.
echo 'deb [arch=armhf] http://ftp.debian.org/debian buster-backports main' >> /etc/apt/sources.list
apt-get install -y qemu-user-static=5.0-14~ bpo10+1

Reply:
Hi @qiluo-msft, @Sabareesh-Kumar-Anandan,

   We actually can't upgrade the qemu-user-static by doing _apt-get install -y qemu-user-static=5.0-14~ bpo10+1_. Because the docker sonic-slave-buster is already running in the emulated arm environment, if you try to install qemu-user-static, it will pull the qemu-user-static binaries from the Debian repo for the arm arch and these binaries are to emulate something else from the arm host arch.

   What we want is to update a qemu-arm-static binary emulating the arm client arch from the x86_64 host arch, copying the new version from multiarch/qemu-user-static:x86_64-arm-5.0.0-2 is a correct way to do this.

Thanks,
Benjamin